src/ui/components/button/BaseButton.ts

```diff
@@ -0,0 +1,116 @@
+import { Container, Polygon, Sprite, Texture } from "pixi.js";
+import { ButtonBackground, ButtonState, ButtonStates, Cursor } from "./BaseButton.types";
+import { Events } from "../../../common/Events";
+
+/**
+ * The base button component.
+ * 
+ * The constructor expects a background texture to initialise the component, also accepts two optional parameters.
+ */
+export class BaseButton extends Container {
```

Comment:
`Button` will be enough as a base class. No need to say `Base` as prefix

Reply:
updated as suggested;

https://github.com/irgat/soloTest/pull/3/commits/dd48dd0809bf5f9bc052d6e70a2ac52a1c2e4e25